src/TimePicker/TimePicker.tsx

```diff
@@ -58,28 +53,19 @@ const getComponents = (
     : {};
 };
 
-const Props = t.refinement(
-  t.struct({
-    onChange: t.Function,
-    value: t.maybe(Time),
-    minTime: t.maybe(Time),
-    maxTime: t.maybe(Time),
-    placeholder: t.maybe(t.String),
-    timeFormat: t.maybe(TimeFormat),
-    timeFormatter: t.maybe(t.Function),
-    searchable: t.Boolean,
-    size: t.enums.of(["medium", "small"]),
-    id: t.maybe(t.String),
-    className: t.maybe(t.String),
-    style: t.maybe(t.Object),
-    menuPosition: t.enums.of(["bottom", "top"]),
-    disabled: t.maybe(t.Boolean)
-  }),
-  ({ value, minTime, maxTime }: any) =>
-    lteTime(minTime, maxTime) &&
-    (!value || (lteTime(value, maxTime) && lteTime(minTime, value))),
-  "Props"
-);
+function validProps({ value, minTime, maxTime }: TimePicker.Props): boolean {
+  const validTimes =
+    (value == null || isTime(value)) &&
+    (minTime == null || isTime(minTime)) &&
+    (maxTime == null || isTime(maxTime));
+  const minMaxCoherent =
+    minTime != null && maxTime != null && lteTime(minTime, maxTime);
```

Comment:
Shouldn't this be `minTime == null || maxTime == null || lteTime(minTime, maxTime)`?

Reply:
yup, good catch, thanks